### DISCUSSION TITLE
Search Filter Webpart sort accented characters

### DISCUSSION THREAD

**Discussion by felix-antoine-dion** [microsoft-search/pnp-modern-search#3092] [Q&A] [answered]
**Version used**
4.8

**Describe the bug**
I have contents that contains a refinable string with these possible values: "Testa", "test", "téa".
I use a french ui and would expect the filter's values to be sorted this way: 

1. `téa`,
1. test, 
1. Testa

...but I get the following values instead: 

1. test
1. Testa
1. `téa`.

**To Reproduce**
1. Configure a Search Result WebPart and Filter WebPart.
1. Configure the Filter WebPart to use a Refinable String that will contain accented character with the following available values: "Testa", "test", "téa"

**Expected behavior**
In french, I would expect the values to be sorted "téa", "test", "Testa". Basicaly, accents should not influence the sort result.

**Desktop:**
Edge 114.0.1823.37
Windows 10

**Additional context**
I've searched a bit and found out this what seems to be the faulty line
https://github.com/microsoft-search/pnp-modern-search/blob/4168174e35b6f68b28fd2ce52a8efbe5bbbef03a/search-parts/src/webparts/searchFilters/SearchFiltersWebPart.ts#L1200

We could replace the lowdash's sortBy function by the native Array's sort function and the use of Intl.Collator().compare. This function enables the sorting to be executed based on the current local of the user.This function seems to be supported by most recent browser: https://caniuse.com/?search=Intl.Collator

The line could be rewritten this way:

```typescript
const collator = new Intl.Collator();
const sortedName = filter.values.sort((a, b) => collator.compare(a.name, b.name));
filter.values = sortDirection === FilterSortDirection.Ascending ? sortedName  : sortedName.reverse(); 
```

**Comment by wobba** [comment, marked as answer]
Try with 4.9 as we changed sorting to use toLocaleLowerCase when comparing.

**Reply by felix-antoine-dion** [reply]
Thank you Mikael, this effectively fixed the issue. Thanks again for this terrific webpart.
